app/models/role.rb

```diff
@@ -4,4 +4,12 @@ class Role < ApplicationRecord
   has_many :users, dependent: :restrict_with_exception
 
   validates :name, presence: true
+
+  before_validation :set_random_color, on: :create
+
+  private
+
+  def set_random_color
+    self.color = "##{SecureRandom.hex(3)}"
```

Comment:
To avoid collisions, isn't 6 char hex better?

Reply:
What do you mean?
`SecureRandom.hex(3)` generates 6 hex chars random.
It's a bit deceiving but the parameter isn't the length but the byteLength.

Reply:
:facepalm: